robottelo/populate/main.py

```diff
@@ -0,0 +1,58 @@
+# coding: utf8
+"""Point of entry for populate and validate used in scripts"""
+import import_string
+import os
+import yaml
+
+from robottelo.populate.constants import DEFAULT_CONFIG
+
+base_path = os.path.join('tests', 'foreman', 'data')
+
+
+def load_data(datafile):
+    """Loads YAML file as a dictionary"""
+    if datafile.endswith(('.yml', '.yaml', 'json')):
+        if not datafile.startswith('/'):
+            datafile = os.path.join(base_path, datafile)
+        with open(datafile) as datafile:
+            return yaml.load(datafile)
+    return yaml.load(datafile)
+
+
+def get_populator(data, verbose):
+    """Gets an instance of populator dynamically"""
+    if not isinstance(data, dict):
+        data = load_data(data)
+
+    config = DEFAULT_CONFIG.copy()
+    config.update(data.get('config', {}))
+    verbose = verbose or config.get('verbose')
+
+    populator_name = config['populator']
+    populator_module_name = config['populators'][populator_name]['module']
+    populator_class = import_string(populator_module_name)
+    populator = populator_class(data=data, verbose=verbose)
+    return populator
+
+
+def populate(data, **extra_options):
+    """Loads and execute populator in populate mode"""
+    verbose = extra_options.get('verbose')
+    populator = get_populator(data, verbose)
+    populator.execute(mode='populate')
+    populator.logger.info("Populator finished!")
+    return populator
+
+
+def validate(data, **extra_options):
+    """Loads and execute populator in validate mode"""
+    verbose = extra_options.get('verbose')
+    populator = get_populator(data, verbose)
+    populator.execute(mode='validate')
+    populator.logger.info("Validator finished!")
+    return populator
+
+
+def wrap_context(context):
```

Comment:
I forgot to write the code for `wrap_context` that function existis for a reason, it should wrap the result in an object accessible via keys and attributes (a dict with getattr implemented)